src/goto-instrument/code_contracts.cpp

```diff
@@ -1595,8 +1615,15 @@ exprt assigns_clauset::compatible_expression(
     {
       if(first_iter)
       {
-        current_target_compatible =
-          target->compatible_expression(*called_target);
+        // TODO: generalize this
+        // currently only supports pointers
```

Comment:
Could you turn this comment into a INVARIANT and open a GitHub issue to track this case?

Reply:
I was obliged to do an if statement instead. Otherwise, the regression tests for structs and arrays would fail.